SeedManager is hardcoded to always look for .ts files, even without ts-node

**Describe the bug**
Attempting to run a seeder from the CLI will always look for a `.ts` file, regardless of whether `useTsNode` / `tsNode` options are set to false. 

https://github.com/mikro-orm/mikro-orm/blob/a0a19c22604586c1f3256aba7759c3204e1f02b0/packages/seeder/src/SeedManager.ts#L49

This makes it impossible to run seeders in production from `dist`, without ts-node.

**Stack trace**
```
Error: Cannot find module 'D:/gabu/server/dist/seeders/database.seeder.ts'
Require stack:
- D:\gabu\server\node_modules\@mikro-orm\seeder\SeedManager.js
- D:\gabu\server\node_modules\@mikro-orm\seeder\index.js
- D:\gabu\server\node_modules\@mikro-orm\core\MikroORM.js
- D:\gabu\server\node_modules\@mikro-orm\core\index.js
- D:\gabu\server\node_modules\@mikro-orm\cli\CLIHelper.js
- D:\gabu\server\node_modules\@mikro-orm\cli\cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at D:\gabu\server\node_modules\@mikro-orm\seeder\SeedManager.js:46:76
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async SeedManager.seedString (D:\gabu\server\node_modules\@mikro-orm\seeder\SeedManager.js:46:28)
    at async Object.handler (D:\gabu\server\node_modules\@mikro-orm\cli\commands\DatabaseSeedCommand.js:26:9) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\gabu\\server\node_modules\\@mikro-orm\\seeder\\SeedManager.js',
    'D:\\gabu\\server\node_modules\\@mikro-orm\\seeder\\index.js',
    'D:\\gabu\\server\node_modules\\@mikro-orm\\core\\MikroORM.js',
    'D:\\gabu\\server\node_modules\\@mikro-orm\\core\\index.js',
    'D:\\gabu\\server\node_modules\\@mikro-orm\\cli\\CLIHelper.js',
    'D:\\gabu\\server\node_modules\\@mikro-orm\\cli\\cli.js'
  ]
}
```

**To Reproduce**
Thanks to the new `Utils.detectTsNode()` it's very easy to use `src/seeders` in development and `dist/seeders` in production:
```ts
export default {
    // ...
    seeder: {
        path: Utils.detectTsNode() ? './src/seeders' : './dist/seeders',
        defaultSeeder: 'DatabaseSeeder',
    },
    // ...
}
```

This makes it possible to create new seeders in `src` (with `useTsNode = true`):
```ts
yarn mikro-orm seeder:create DatabaseSeeder
yarn run v1.22.5
$ D:\gabu\server\node_modules\.bin\mikro-orm seeder:create DatabaseSeeder
Seeder DatabaseSeeder successfully created at D:/gabu/server/src/seeders/database.seeder.ts
```

However after running `tsc` and then trying to run this seeder in production from `dist` (with `useTsNode = false`) results in the SeedManager trying to look for a `.ts` file in `dist`, which obviously fails to find anything.

```bash
yarn cross-env NODE_ENV=production mikro-orm seeder:run
$ D:\gabu\server\node_modules\.bin\cross-env NODE_ENV=production mikro-orm seeder:run
mikro-orm seeder:run

Seed the database using the seeder class

Options:
  -c, --class    Seeder class to run                                    [string]
  -v, --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

Error: Cannot find module 'D:/gabu/server/dist/seeders/database.seeder.ts'
```

Since the `.ts` file extension is hardcoded in SeedManager there is no workaround for this.

**Expected behavior**
The seeder CLI command should look for both `.js` and `.ts` files on the given seeder path, similar to how migrations now work. Replacing the hardcode extension with a glob pattern would probably solve this issue.

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | v16.13.2 |
| typescript | 4.5.5 |
| mikro-orm | 5.0.0 |
| your-driver | postgres |

--- a/docs/docs/configuration.md
+++ b/docs/docs/configuration.md
@@ -607,4 +607,7 @@ Full list of supported options:
 | `MIKRO_ORM_SCHEMA_GENERATOR_DISABLE_FOREIGN_KEYS` | `migrations.disableForeignKeys` |
 | `MIKRO_ORM_SCHEMA_GENERATOR_CREATE_FOREIGN_KEY_CONSTRAINTS` | `migrations.createForeignKeyConstraints` |
 | `MIKRO_ORM_SEEDER_PATH` | `seeder.path` |
+| `MIKRO_ORM_SEEDER_PATH_TS` | `seeder.pathTs` |
+| `MIKRO_ORM_SEEDER_GLOB` | `seeder.glob` |
+| `MIKRO_ORM_SEEDER_EMIT` | `seeder.emit` |
 | `MIKRO_ORM_SEEDER_DEFAULT_SEEDER` | `seeder.defaultSeeder` |

--- a/docs/docs/seeding.md
+++ b/docs/docs/seeding.md
@@ -7,18 +7,31 @@ script or combine multiple seed scripts.
 
 ## Configuration
 
+> `seeder.path` and `seeder.pathTs` works the same way as `entities` and
+> `entitiesTs` in entity discovery.
+
 The seeder has a few default settings that can be changed easily through the MikroORM config. Underneath you find the configuration options with their defaults.
 
 ```ts
 MikroORM.init({
   seeder: {
     path: './seeders', // path to the folder with seeders
-    defaultSeeder: 'DatabaseSeeder' // default seeder class name
-  }
+    pathTs: undefined, // path to the folder with TS seeders (if used, we should put path to compiled files in `path`)
+    defaultSeeder: 'DatabaseSeeder', // default seeder class name
+    glob: '!(*.d).{js,ts}', // how to match seeder files (all .js and .ts files, but not .d.ts)
+    emit: 'ts', // seeder generation mode
+    fileName: (className: string) => className, // seeder file naming convention
+  },
 });
 ```
 
-You can also override these default using the [environment variables](configuration.md#using-environment-variables) `MIKRO_ORM_SEEDER_PATH` and `MIKRO_ORM_SEEDER_DEFAULT_SEEDER`.
+We can also override these default using the [environment variables](configuration.md#using-environment-variables):
+
+- `MIKRO_ORM_SEEDER_PATH`
+- `MIKRO_ORM_SEEDER_PATH_TS`
+- `MIKRO_ORM_SEEDER_EMIT`
+- `MIKRO_ORM_SEEDER_GLOB`
+- `MIKRO_ORM_SEEDER_DEFAULT_SEEDER`.
 
 ## Seeders
 
@@ -266,3 +279,27 @@ afterAll(async () => {
   await orm.close();
 });
 ```
+
+## Running migrations in production
+
+In production environment we might want to use compiled seeder files. All we need to do is to configure the seeder path accordingly:
+
+```ts
+import { MikroORM, Utils } from '@mikro-orm/core';
+
+await MikroORM.init({
+  seeder: {
+    path: 'dist/seeders',
+    pathTs: 'src/seeders',
+  },
+  // or alternatively
+  // seeder: {
+  //   path: Utils.detectTsNode() ? 'src/seeders' : 'dist/seeders',
+  // },
+  // ...
+});
+```
+
+This should allow using CLI to generate TS seeder files (as in CLI we probably
+have TS support enabled), while using compiled JS files in production, where ts-node
+is not registered.

--- a/docs/versioned_docs/version-5.0/configuration.md
+++ b/docs/versioned_docs/version-5.0/configuration.md
@@ -607,4 +607,7 @@ Full list of supported options:
 | `MIKRO_ORM_SCHEMA_GENERATOR_DISABLE_FOREIGN_KEYS` | `migrations.disableForeignKeys` |
 | `MIKRO_ORM_SCHEMA_GENERATOR_CREATE_FOREIGN_KEY_CONSTRAINTS` | `migrations.createForeignKeyConstraints` |
 | `MIKRO_ORM_SEEDER_PATH` | `seeder.path` |
+| `MIKRO_ORM_SEEDER_PATH_TS` | `seeder.pathTs` |
+| `MIKRO_ORM_SEEDER_GLOB` | `seeder.glob` |
+| `MIKRO_ORM_SEEDER_EMIT` | `seeder.emit` |
 | `MIKRO_ORM_SEEDER_DEFAULT_SEEDER` | `seeder.defaultSeeder` |

--- a/docs/versioned_docs/version-5.0/seeding.md
+++ b/docs/versioned_docs/version-5.0/seeding.md
@@ -7,18 +7,31 @@ script or combine multiple seed scripts.
 
 ## Configuration
 
+> `seeder.path` and `seeder.pathTs` works the same way as `entities` and
+> `entitiesTs` in entity discovery.
+
 The seeder has a few default settings that can be changed easily through the MikroORM config. Underneath you find the configuration options with their defaults.
 
 ```ts
 MikroORM.init({
   seeder: {
     path: './seeders', // path to the folder with seeders
-    defaultSeeder: 'DatabaseSeeder' // default seeder class name
-  }
+    pathTs: undefined, // path to the folder with TS seeders (if used, we should put path to compiled files in `path`)
+    defaultSeeder: 'DatabaseSeeder', // default seeder class name
+    glob: '!(*.d).{js,ts}', // how to match seeder files (all .js and .ts files, but not .d.ts)
+    emit: 'ts', // seeder generation mode
+    fileName: (className: string) => className, // seeder file naming convention
+  },
 });
 ```
 
-You can also override these default using the [environment variables](configuration.md#using-environment-variables) `MIKRO_ORM_SEEDER_PATH` and `MIKRO_ORM_SEEDER_DEFAULT_SEEDER`.
+We can also override these default using the [environment variables](configuration.md#using-environment-variables):
+
+- `MIKRO_ORM_SEEDER_PATH`
+- `MIKRO_ORM_SEEDER_PATH_TS`
+- `MIKRO_ORM_SEEDER_EMIT`
+- `MIKRO_ORM_SEEDER_GLOB`
+- `MIKRO_ORM_SEEDER_DEFAULT_SEEDER`.
 
 ## Seeders
 
@@ -266,3 +279,27 @@ afterAll(async () => {
   await orm.close();
 });
 ```
+
+## Running migrations in production
+
+In production environment we might want to use compiled seeder files. All we need to do is to configure the seeder path accordingly:
+
+```ts
+import { MikroORM, Utils } from '@mikro-orm/core';
+
+await MikroORM.init({
+  seeder: {
+    path: 'dist/seeders',
+    pathTs: 'src/seeders',
+  },
+  // or alternatively
+  // seeder: {
+  //   path: Utils.detectTsNode() ? 'src/seeders' : 'dist/seeders',
+  // },
+  // ...
+});
+```
+
+This should allow using CLI to generate TS seeder files (as in CLI we probably
+have TS support enabled), while using compiled JS files in production, where ts-node
+is not registered.

--- a/packages/cli/src/commands/CreateSeederCommand.ts
+++ b/packages/cli/src/commands/CreateSeederCommand.ts
@@ -1,7 +1,5 @@
 import type { Arguments, Argv, CommandModule } from 'yargs';
-import type { MikroORM } from '@mikro-orm/core';
 import { colors } from '@mikro-orm/core';
-import type { AbstractSqlDriver } from '@mikro-orm/knex';
 import { CLIHelper } from '../CLIHelper';
 
 export class CreateSeederCommand<T> implements CommandModule<T, { seeder: string }> {
@@ -20,28 +18,22 @@ export class CreateSeederCommand<T> implements CommandModule<T, { seeder: string
    * @inheritDoc
    */
   async handler(args: Arguments<{ seeder: string }>) {
-    const seederName = CreateSeederCommand.getSeederClassName(args.seeder);
-    const orm = await CLIHelper.getORM(undefined) as MikroORM<AbstractSqlDriver>;
+    const className = CreateSeederCommand.getSeederClassName(args.seeder);
+    const orm = await CLIHelper.getORM();
     const seeder = orm.getSeeder();
-    const path = await seeder.createSeeder(seederName);
+    const path = await seeder.createSeeder(className);
     CLIHelper.dump(colors.green(`Seeder ${args.seeder} successfully created at ${path}`));
     await orm.close(true);
   }
 
   /**
    * Will return a seeder name that is formatted like this EntitySeeder
-   * @param seedName
-   * @private
    */
-  static getSeederClassName(seedName: string): string {
-    const seedNameMatches = seedName.match(/(.+)(seeder)/i);
-    if (seedNameMatches) {
-      seedName = seedNameMatches[1];
-    }
-    const seedNameSplit = seedName.split('-');
-    return seedNameSplit.map(name => {
-      return name.charAt(0).toUpperCase() + name.slice(1);
-    }).join('') + 'Seeder';
+  private static getSeederClassName(name: string): string {
+    name = name.match(/(.+)seeder/i)?.[1] ?? name;
+    const parts = name.split('-');
+
+    return parts.map(name => name.charAt(0).toUpperCase() + name.slice(1)).join('') + 'Seeder';
   }
 
 }

--- a/packages/cli/src/commands/DatabaseSeedCommand.ts
+++ b/packages/cli/src/commands/DatabaseSeedCommand.ts
@@ -1,7 +1,5 @@
 import type { Arguments, Argv, CommandModule } from 'yargs';
-import type { MikroORM } from '@mikro-orm/core';
 import { colors } from '@mikro-orm/core';
-import type { AbstractSqlDriver } from '@mikro-orm/knex';
 import { CLIHelper } from '../CLIHelper';
 
 export class DatabaseSeedCommand<T> implements CommandModule<T, { class: string }> {
@@ -21,11 +19,10 @@ export class DatabaseSeedCommand<T> implements CommandModule<T, { class: string 
    * @inheritDoc
    */
   async handler(args: Arguments<{ class?: string }>) {
-    const orm = await CLIHelper.getORM(undefined) as MikroORM<AbstractSqlDriver>;
-    const seeder = orm.getSeeder();
-    const seederClass = args.class || orm.config.get('seeder').defaultSeeder;
-    await seeder.seedString(seederClass);
-    CLIHelper.dump(colors.green(`Seeder ${seederClass} successfully seeded`));
+    const orm = await CLIHelper.getORM();
+    const className = args.class ?? orm.config.get('seeder').defaultSeeder!;
+    await orm.getSeeder().seedString(className);
+    CLIHelper.dump(colors.green(`Seeder ${className} successfully executed`));
     await orm.close(true);
   }
 

--- a/packages/cli/src/commands/MigrationCommandFactory.ts
+++ b/packages/cli/src/commands/MigrationCommandFactory.ts
@@ -178,7 +178,7 @@ export class MigrationCommandFactory {
 
     if (args.seed !== undefined) {
       const seeder = orm.getSeeder();
-      const seederClass = args.seed || orm.config.get('seeder').defaultSeeder;
+      const seederClass = args.seed || orm.config.get('seeder').defaultSeeder!;
       await seeder.seedString(seederClass);
       CLIHelper.dump(colors.green(`Database seeded successfully with seeder class ${seederClass}`));
     }

--- a/packages/cli/src/commands/SchemaCommandFactory.ts
+++ b/packages/cli/src/commands/SchemaCommandFactory.ts
@@ -113,7 +113,7 @@ export class SchemaCommandFactory {
 
     if (args.seed !== undefined) {
       const seeder = orm.getSeeder();
-      await seeder.seedString(args.seed || orm.config.get('seeder').defaultSeeder);
+      await seeder.seedString(args.seed || orm.config.get('seeder').defaultSeeder!);
     }
 
     CLIHelper.dump(colors.green(successMessage));

--- a/packages/core/src/typings.ts
+++ b/packages/core/src/typings.ts
@@ -666,9 +666,10 @@ export interface HydratorConstructor {
 }
 
 export interface ISeedManager {
-  seed(...seederClasses: { new(): Seeder }[]): Promise<void>;
-  seedString(...seederClasses: string[]): Promise<void>;
-  createSeeder(seederClass: string): Promise<string>;
+  seed(...classNames: Constructor<Seeder>[]): Promise<void>;
+  /** @internal */
+  seedString(...classNames: string[]): Promise<void>;
+  createSeeder(className: string): Promise<string>;
 }
 
 export interface Seeder {

--- a/packages/core/src/utils/Configuration.ts
+++ b/packages/core/src/utils/Configuration.ts
@@ -110,6 +110,9 @@ export class Configuration<D extends IDatabaseDriver = IDatabaseDriver> {
     seeder: {
       path: './seeders',
       defaultSeeder: 'DatabaseSeeder',
+      glob: '!(*.d).{js,ts}',
+      emit: 'ts',
+      fileName: (className: string) => className,
     },
   };
 
@@ -373,6 +376,15 @@ export type MigrationsOptions = {
   migrationsList?: MigrationObject[];
 };
 
+export type SeederOptions = {
+  path?: string;
+  pathTs?: string;
+  glob?: string;
+  defaultSeeder?: string;
+  emit?: 'js' | 'ts';
+  fileName?: (className: string) => string;
+};
+
 export interface PoolConfig {
   name?: string;
   afterCreate?: (...a: unknown[]) => unknown;
@@ -461,7 +473,7 @@ export interface MikroORMOptions<D extends IDatabaseDriver = IDatabaseDriver> ex
     options?: Dictionary;
   };
   metadataProvider: { new(config: Configuration): MetadataProvider };
-  seeder: { path: string; defaultSeeder: string };
+  seeder: SeederOptions;
 }
 
 export type Options<D extends IDatabaseDriver = IDatabaseDriver> =

--- a/packages/core/src/utils/ConfigurationLoader.ts
+++ b/packages/core/src/utils/ConfigurationLoader.ts
@@ -202,6 +202,9 @@ export class ConfigurationLoader {
 
     ret.seeder = {};
     read(ret.seeder, 'MIKRO_ORM_SEEDER_PATH', 'path');
+    read(ret.seeder, 'MIKRO_ORM_SEEDER_PATH_TS', 'pathTs');
+    read(ret.seeder, 'MIKRO_ORM_SEEDER_GLOB', 'glob');
+    read(ret.seeder, 'MIKRO_ORM_SEEDER_EMIT', 'emit');
     read(ret.seeder, 'MIKRO_ORM_SEEDER_DEFAULT_SEEDER', 'defaultSeeder');
     cleanup(ret, 'seeder');
 

--- a/packages/seeder/src/SeedManager.ts
+++ b/packages/seeder/src/SeedManager.ts
@@ -1,4 +1,5 @@
-import type { ISeedManager, EntityManager } from '@mikro-orm/core';
+import globby from 'globby';
+import type { Constructor, EntityManager, ISeedManager } from '@mikro-orm/core';
 import { Utils } from '@mikro-orm/core';
 import type { Seeder } from './Seeder';
 import { ensureDir, writeFile } from 'fs-extra';
@@ -7,61 +8,81 @@ export class SeedManager implements ISeedManager {
 
   private readonly config = this.em.config;
   private readonly options = this.config.get('seeder');
-  private readonly absolutePath = Utils.absolutePath(this.options.path, this.config.get('baseDir'));
+  private readonly absolutePath: string;
 
   constructor(private readonly em: EntityManager) {
     this.em = this.em.fork();
     this.config.set('persistOnCreate', true);
+    /* istanbul ignore next */
+    const key = (this.config.get('tsNode', Utils.detectTsNode()) && this.options.pathTs) ? 'pathTs' : 'path';
+    this.absolutePath = Utils.absolutePath(this.options[key]!, this.config.get('baseDir'));
   }
 
-  async seed(...seederClasses: { new(): Seeder }[]): Promise<void> {
-    for (const seederClass of seederClasses) {
-      const seeder = new seederClass();
+  async seed(...classNames: Constructor<Seeder>[]): Promise<void> {
+    for (const SeederClass of classNames) {
+      const seeder = new SeederClass();
       await seeder.run(this.em);
       await this.em.flush();
       this.em.clear();
     }
   }
 
-  async seedString(...seederClasses: string[]): Promise<void> {
-    for (const seederClass of seederClasses) {
-      const seedersPath = Utils.normalizePath(this.absolutePath);
-      const filePath = `${seedersPath}/${this.getFileName(seederClass)}`;
-      const seeder = await import(filePath);
-      await this.seed(seeder[seederClass]);
+  /**
+   * @internal
+   */
+  async seedString(...classNames: string[]): Promise<void> {
+    const path = `${this.absolutePath}/${this.options.glob}`;
+    const files = await globby(path);
+    const classMap = new Map<string, Constructor<Seeder>>();
+
+    for (const path of files) {
+      const exports = await Utils.dynamicImport(path);
+
+      for (const name of Object.keys(exports)) {
+        classMap.set(name, exports[name]);
+      }
+    }
+
+    for (const className of classNames) {
+      const seederClass = classMap.get(className);
+
+      if (!seederClass) {
+        throw new Error(`Seeder class ${className} not found in ${Utils.relativePath(path, process.cwd())}`);
+      }
+
+      await this.seed(seederClass);
     }
   }
 
-  async createSeeder(seederClass: string): Promise<string> {
+  async createSeeder(className: string): Promise<string> {
     await this.ensureSeedersDirExists();
-    return this.generate(seederClass);
-  }
-
-  private getFileName(seederClass: string): string {
-    const split = seederClass.split(/(?=[A-Z])/);
-    const parts = split.reduce((previousValue: string[], currentValue: string, index: number) => {
-      if (index === split.length - 1) {
-        return previousValue;
-      }
-      previousValue.push(currentValue.toLowerCase());
-      return previousValue;
-    }, []);
-    return `${parts.join('-')}.seeder.ts`;
+    return this.generate(className);
   }
 
   private async ensureSeedersDirExists() {
-    await ensureDir(Utils.normalizePath(this.absolutePath));
+    await ensureDir(this.absolutePath);
   }
 
-  private async generate(seederClass: string): Promise<string> {
-    const seedersPath = Utils.normalizePath(this.absolutePath);
-    const filePath = `${seedersPath}/${this.getFileName(seederClass)}`;
+  private async generate(className: string): Promise<string> {
+    const fileName = `${this.options.fileName!(className)}.${this.options.emit}`;
+    const filePath = `${this.absolutePath}/${fileName}`;
+    let ret = '';
 
-    let ret = `import type { EntityManager } from '@mikro-orm/core';\n`;
-    ret += `import { Seeder } from '@mikro-orm/seeder';\n\n`;
-    ret += `export class ${seederClass} extends Seeder {\n\n`;
-    ret += `  async run(em: EntityManager): Promise<void> {}\n\n`;
-    ret += `}\n`;
+    if (this.options.emit === 'ts') {
+      ret += `import type { EntityManager } from '@mikro-orm/core';\n`;
+      ret += `import { Seeder } from '@mikro-orm/seeder';\n\n`;
+      ret += `export class ${className} extends Seeder {\n\n`;
+      ret += `  async run(em: EntityManager): Promise<void> {}\n\n`;
+      ret += `}\n`;
+    } else {
+      ret += `'use strict';\n`;
+      ret += `Object.defineProperty(exports, '__esModule', { value: true });\n`;
+      ret += `const { Seeder } = require('@mikro-orm/seeder');\n\n`;
+      ret += `class ${className} extends Seeder {\n\n`;
+      ret += `  async run(em: EntityManager): Promise<void> {}\n\n`;
+      ret += `}\n`;
+      ret += `exports.${className} = ${className};\n`;
+    }
 
     await writeFile(filePath, ret);
 

--- a/packages/seeder/src/Seeder.ts
+++ b/packages/seeder/src/Seeder.ts
@@ -5,11 +5,8 @@ export abstract class Seeder {
   abstract run(em: EntityManager): Promise<void>;
 
   protected call(em: EntityManager, seeders: { new(): Seeder }[]): Promise<void> {
-    return new Promise((resolve, reject) => {
-      Promise.all(seeders.map(s => {
-        return (new s()).run(em.fork());
-      })).then(() => resolve()).catch(reject);
-    });
+    const promises = seeders.map(s => (new s()).run(em.fork()));
+    return Promise.all(promises).then();
   }
 
 }

--- a/tests/features/cli/CreateSeederCommand.test.ts
+++ b/tests/features/cli/CreateSeederCommand.test.ts
@@ -43,6 +43,7 @@ describe('CreateSeederCommand', () => {
 
   test('should generate seeder class with all kind of names', async () => {
     const cmd = new CreateSeederCommand();
+    // @ts-expect-error private method
     const spy = jest.spyOn(CreateSeederCommand, 'getSeederClassName');
     await cmd.handler({ seeder: 'DatabaseSeeder' } as any);
     expect(spy).toHaveLastReturnedWith('DatabaseSeeder');

--- a/tests/features/seeder/factory.test.ts
+++ b/tests/features/seeder/factory.test.ts
@@ -1,6 +1,5 @@
 import { MikroORM } from '@mikro-orm/core';
 import { Factory } from '@mikro-orm/seeder';
-import type { SqliteDriver } from '@mikro-orm/sqlite';
 import type { Faker } from '@mikro-orm/seeder';
 import { House } from './entities/house.entity';
 import { Project } from './entities/project.entity';
@@ -34,7 +33,7 @@ export class HouseFactory extends Factory<House> {
 
 describe('Factory', () => {
 
-  let orm: MikroORM<SqliteDriver>;
+  let orm: MikroORM;
   let persistSpy: SpyInstance;
   let flushSpy: SpyInstance;
 

--- a/tests/features/seeder/run-seeders.test.ts
+++ b/tests/features/seeder/run-seeders.test.ts
@@ -1,5 +1,4 @@
 import { MikroORM } from '@mikro-orm/core';
-import type { SqliteDriver } from '@mikro-orm/sqlite';
 import { House } from './entities/house.entity';
 import { Project } from './entities/project.entity';
 import { User } from './entities/user.entity';
@@ -7,7 +6,7 @@ import { DatabaseSeeder } from '../../database/seeder/database.seeder';
 
 describe('Run seeders', () => {
 
-  let orm: MikroORM<SqliteDriver>;
+  let orm: MikroORM;
 
   beforeAll(async () => {
     orm = await MikroORM.init({
@@ -21,7 +20,6 @@ describe('Run seeders', () => {
   afterAll(() => orm.close(true));
 
   test('that by calling DatabaseSeeder both ProjectSeeder and UserSeeder have been called', async () => {
-
     const seeder = new DatabaseSeeder();
     await seeder.run(orm.em);
 
@@ -31,4 +29,5 @@ describe('Run seeders', () => {
     const users = await orm.em.findAndCount(User, {});
     expect(users[1]).toBe(1);
   });
+
 });

--- a/tests/features/seeder/seed-manager.test.ts
+++ b/tests/features/seeder/seed-manager.test.ts
@@ -1,32 +1,29 @@
-import SpyInstance = jest.SpyInstance;
-
+import { remove, readFile } from 'fs-extra';
 import { MikroORM } from '@mikro-orm/core';
 import { SeedManager } from '@mikro-orm/seeder';
-import type { SqliteDriver } from '@mikro-orm/sqlite';
 import { SchemaGenerator } from '@mikro-orm/sqlite';
-// noinspection ES6PreferShortImport
 import { initORMSqlite } from '../../bootstrap';
 import { Book3Seeder } from '../../database/seeder/book3.seeder';
-import { remove, readFile } from 'fs-extra';
 import { Author3Seeder } from '../../database/seeder/author3.seeder';
 
-const createSchema = jest.spyOn(SchemaGenerator.prototype, 'createSchema');
-createSchema.mockImplementation(async () => void 0);
-const dropSchema = jest.spyOn(SchemaGenerator.prototype, 'dropSchema');
-dropSchema.mockImplementation(async () => void 0);
+describe('Seeder', () => {
 
-describe('MikroOrmSeeder', () => {
-
-  let orm: MikroORM<SqliteDriver>;
-  let getORMMock: SpyInstance;
+  let orm: MikroORM;
 
   beforeAll(async () => {
     orm = await initORMSqlite();
-    getORMMock = jest.spyOn(MikroORM, 'init');
+    const getORMMock = jest.spyOn(MikroORM, 'init');
     getORMMock.mockResolvedValue(orm);
+    const createSchema = jest.spyOn(SchemaGenerator.prototype, 'createSchema');
+    createSchema.mockImplementation();
+    const dropSchema = jest.spyOn(SchemaGenerator.prototype, 'dropSchema');
+    dropSchema.mockImplementation();
   });
 
-  afterAll(async () => await orm.close(true));
+  afterAll(async () => {
+    await orm.close(true);
+    jest.restoreAllMocks();
+  });
 
   test('seed', async () => {
     const seeder = orm.getSeeder();
@@ -43,7 +40,10 @@ describe('MikroOrmSeeder', () => {
   });
 
   test('seedString', async () => {
-    orm.config.set('seeder', { path: './database/seeder', defaultSeeder: 'DatabaseSeeder' });
+    const options = orm.config.get('seeder');
+    options.path = './database/seeder';
+    options.defaultSeeder = 'DatabaseSeeder';
+    orm.config.set('seeder', options);
     const seeder = orm.getSeeder();
     const seedMock = jest.spyOn(SeedManager.prototype, 'seed');
 
@@ -51,16 +51,38 @@ describe('MikroOrmSeeder', () => {
     expect(seedMock).toHaveBeenCalledTimes(1);
     await seeder.seedString('Book3Seeder', 'Author3Seeder');
     expect(seedMock).toHaveBeenCalledTimes(3);
+
+    const re = 'Seeder class Unknown not found in ./tests/database/seeder/!(*.d).{js,ts}';
+    await expect(seeder.seedString('Unknown')).rejects.toThrow(re);
   });
 
-  test('createSeeder', async () => {
-    orm.config.set('seeder', { path: process.cwd() + '/temp/seeders', defaultSeeder: 'DatabaseSeeder' });
+  test('createSeeder (TS)', async () => {
+    const options = orm.config.get('seeder');
+    options.path = process.cwd() + '/temp/seeders';
+    options.defaultSeeder = 'DatabaseSeeder';
+    orm.config.set('seeder', options);
     const seeder = orm.getSeeder();
 
     const seederFile = await seeder.createSeeder('Publisher3Seeder');
-    expect(seederFile).toBe(process.cwd() + `/temp/seeders/publisher3.seeder.ts`);
+    expect(seederFile).toBe(process.cwd() + `/temp/seeders/Publisher3Seeder.ts`);
     const fileContents = await readFile(seederFile, 'utf8');
     expect(fileContents).toContain('export class Publisher3Seeder extends Seeder {');
     await remove(seederFile);
   });
+
+  test('createSeeder (JS)', async () => {
+    const options = orm.config.get('seeder');
+    options.path = process.cwd() + '/temp/seeders';
+    options.emit = 'js';
+    options.defaultSeeder = 'DatabaseSeeder';
+    orm.config.set('seeder', options);
+    const seeder = orm.getSeeder();
+
+    const seederFile = await seeder.createSeeder('Publisher3Seeder');
+    expect(seederFile).toBe(process.cwd() + `/temp/seeders/Publisher3Seeder.js`);
+    const fileContents = await readFile(seederFile, 'utf8');
+    expect(fileContents).toContain('exports.Publisher3Seeder = Publisher3Seeder;');
+    await remove(seederFile);
+  });
+
 });

